src/messaging/Message.utils.ts

```diff
@@ -0,0 +1,297 @@
+          body.content.delegationData.isPCR,
+        ],
+        [body.content.signatures.inviter, body.content.signatures.invitee],
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REJECT_ACCEPT_DELEGATION: {
+      const compressedContents: CompressedDelegationData = [
+        body.content.account,
+        body.content.id,
+        body.content.parentId,
+        body.content.permissions,
+        body.content.isPCR,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.INFORM_CREATE_DELEGATION: {
+      const compressedContents: CompressedInformCreateDelegation = [
+        body.content.delegationId,
+        body.content.isPCR,
+      ]
+      return [body.type, compressedContents]
+    }
+    default:
+      throw SDKErrors.ERROR_MESSAGE_BODY_MALFORMED()
+  }
+}
+
+/**
+ * [STATIC] Takes a compressed [[Message]] and decompresses it depending on the message body type.
+ *
+ * @param body The body of the compressed [[Message]] which depends on the [[MessageBodyType]] that needs to be decompressed.
+ *
+ *
+ */
+
+export function decompressMessage(body: CompressedMessageBody): MessageBody {
+  switch (body[0]) {
+    case MessageBodyType.REQUEST_TERMS: {
+      const decompressedContents: IPartialClaim = ClaimUtils.decompress(body[1])
+      return { type: body[0], content: decompressedContents }
+    }
+    case MessageBodyType.SUBMIT_TERMS: {
+      const decompressedContents: ITerms = {
+        claim: ClaimUtils.decompress(body[1][0]),
+        legitimations: body[1][1].map(
+          (attestedClaim: IAttestedClaim | CompressedAttestedClaim) =>
+            !Array.isArray(attestedClaim)
+              ? attestedClaim
+              : AttestedClaimUtils.decompress(attestedClaim)
+        ),
+        delegationId: body[1][2],
+        quote: body[1][3]
+          ? QuoteUtils.decompressAttesterSignedQuote(body[1][3])
+          : undefined,
+        prerequisiteClaims: body[1][4],
+      }
+
+      return { type: body[0], content: decompressedContents }
+    }
+    case MessageBodyType.REJECT_TERMS: {
+      const decompressedContents: Pick<
+        ITerms,
+        'claim' | 'legitimations' | 'delegationId'
+      > = {
+        claim: ClaimUtils.decompress(body[1][0]),
+        legitimations: body[1][1].map((val) =>
+          AttestedClaimUtils.decompress(val)
+        ),
+        delegationId: body[1][2] ? body[1][2] : undefined,
+      }
+      return { type: body[0], content: decompressedContents }
+    }
+    case MessageBodyType.REQUEST_ATTESTATION_FOR_CLAIM: {
+      const decompressedContents: IRequestingAttestationForClaim = {
+        requestForAttestation: RequestForAttestationUtils.decompress(
+          body[1][0]
+        ),
+        quote: body[1][1]
+          ? QuoteUtils.decompressQuoteAgreement(body[1][1])
+          : undefined,
+        prerequisiteClaims: body[1][2]
+          ? body[1][2].map((claim) => ClaimUtils.decompress(claim))
```

Comment:
_Info_: This line is currently not covered by unit tests.

Reply:
At the moment we don't use `prerequisiteClaims`. This might turn out to be removed, as it was part of the followup on the quotes. Let us speak with and see what we shall do about it.